File: app/src/main/java/com/codechakra/usergitrepository/GithubApiService.java
package com.codechakra.usergitrepository;

import com.codechakra.usergitrepository.models.Issue;
import com.codechakra.usergitrepository.models.Repos;
import com.codechakra.usergitrepository.models.Users;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;


public interface GithubApiService {
    @GET("users/{login}")
    Call<Users> getUserHistory(@Path("login") String login );

  @GET("users/{login}/repos")
    Call<List<Repos>> getUserRepo(@Path("login") String login);
    //https://api.github.com/repos/octocat/Hello-World/issues{/number}

    @GET("repos/{login}/{reponame}/issues/{number}")
    Call<Issue> getRepoIssue(@Path("login") String login,
                             @Path("reponame") String reponame,
                             @Path("number") String number
                            );
    //octocat/Spoon-Knife/issues
}

File: GitHistory/app/src/main/java/com/adityaoo7/githistory/network/GithubApiService.kt
package com.adityaoo7.githistory.network

import com.adityaoo7.githistory.models.Comment
import com.adityaoo7.githistory.models.Issue
import com.adityaoo7.githistory.models.Repository
import com.adityaoo7.githistory.models.User
import retrofit2.http.GET
import retrofit2.http.Path

const val BASE_URL = "https://api.github.com/"

interface GithubApiService {
    @GET("users/{user}")
    suspend fun getUser(@Path("user") userName: String): User?

    @GET("repos/{user}/{repo}")
    suspend fun getRepository(
        @Path("user") userName: String,
        @Path("repo") repoName: String
    ): Repository?

    @GET("users/{user}/repos")
    suspend fun getRepositories(@Path("user") userName: String): List<Repository>?

    @GET("repos/{user}/{repo}/issues/{number}")
    suspend fun getIssue(
        @Path("user") userName: String,
        @Path("repo") repoName: String,
        @Path("number") issueNumber: Int
    ): Issue?

    @GET("repos/{user}/{repo}/issues")
    suspend fun getIssues(
        @Path("user") userName: String,
        @Path("repo") repoName: String
    ): List<Issue>?

    @GET("repos/{user}/{repo}/issues/{number}/comments")
    suspend fun getComments(
        @Path("user") userName: String,
        @Path("repo") repoName: String,
        @Path("number") issueNumber: Int
    ): List<Comment>?
}